six/modules/c++/six.sicd/source/ComplexData.cpp

```diff
@@ -56,38 +56,42 @@ void ComplexData::getOutputPlaneOffsetAndExtent(
         types::RowCol<size_t>& offset,
         types::RowCol<size_t>& extent) const
 {
-    offset.row = offset.col = 0;
-
     const six::sicd::AreaPlane& areaPlane =
         *radarCollection->area->plane;
+    getOutputPlaneOffsetAndExtent(offset, extent, areaPlane);
+}
+
+void ComplexData::getOutputPlaneOffsetAndExtent(
+        types::RowCol<size_t>& offset,
+        types::RowCol<size_t>& extent,
+        const AreaPlane& areaPlane) const
+{
+    offset.row = offset.col = 0;
     extent.row = areaPlane.xDirection->elements;
     extent.col = areaPlane.yDirection->elements;
 
     if (imageFormation.get())
     {
         const std::string& segmentID =
                 imageFormation->segmentIdentifier;
+        try
+        {
+            const six::sicd::Segment& segment = areaPlane.getSegment(segmentID);
```

Comment:
Rather than a try/catch how about adding a hasSegment () method?  But also, when would this happen and you don't want to error out?  When segmentID is empty?  In that case, would it be better to explicitly check for that?